src/net_processing.cpp

```diff
@@ -750,13 +765,14 @@ static void MaybeSetPeerAsAnnouncingHeaderAndIDs(NodeId nodeid, CConnman& connma
     }
 }
 
-static bool TipMayBeStale(const Consensus::Params &consensusParams) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
+bool PeerManagerImpl::TipMayBeStale() EXCLUSIVE_LOCKS_REQUIRED(cs_main)
 {
     AssertLockHeld(cs_main);
-    if (g_last_tip_update == 0) {
-        g_last_tip_update = GetTime();
+    const Consensus::Params& consensusParams = m_chainparams.GetConsensus();
+    if (m_last_tip_update == 0) {
+        m_last_tip_update = GetTime();
```

Comment:
Now that `m_last_tip_update` isn't a global and can be initialized at the time that `PeerManagerImpl` is constructed, consider initializing it to `GetTime()` in the initializer list and removing this special casing.

Reply:
That seems like a good idea -- I think it'd let you mark `TipMayBeStale` as a const method as well; but I think it goes beyond the mechanical code shuffling that the rest of this PR does.

Reply:
Yep, fine to leave this. There's plenty more to be cleaned up after this PR.